connector/connect/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -40,37 +40,34 @@ message Expression {
 
   message Literal {
     oneof literal_type {
```

Comment:
In catalyst, `Literal` is consist of value and type. The benefit is, there aren't many physical values (timestamp and timestamp ntz are both int64). When we need to add a new data type with existing physical values (e.g. char are varchar), we don't need to update `Literal`.

This is not a design decision made by this PR, so I won't block this PR for this reason.

Reply:
This only works in Scala because of the inherited type hierarchy, the ability of an Any type and deferring the necessary apply / unaply methods.

Since proto does not have this imperative concept of inheritance, we need to embed the physical type with the logical one.